pkg/util/iptables/iptables.go

```diff
@@ -784,3 +786,74 @@ func isResourceError(err error) bool {
 	}
 	return false
 }
+
+type ParseError interface {
+	Line() int
+	Error() string
+}
+
+type parseError struct {
+	cmd  string
+	line int
+}
+
+func (e parseError) Line() int {
+	return e.line
+}
+
+func (e parseError) Error() string {
+	return fmt.Sprintf("%s: input error on line %d: ", e.cmd, e.line)
+}
+
+// ParseRestoreError extracts the line from the error, if it matches returns parseError
+// for example:
+// input err: iptables-restore: line 51 failed
+// output parseError:  cmd = iptables-restore, line = 51
+func ParseRestoreError(err error) (ParseError, bool) {
+	errInfos := strings.Split(err.Error(), ":")
+	if len(errInfos) != 2 {
```

Comment:
I'm afraid of depending on the iptables error message format not changing :/

Reply:
yeah,  this is an uncertainty

Reply:
Just document that, to our best knowledge, this is the output.  If it ever changes we need to update this

Reply:
done